marklogic-data-hub-spark-connector/src/test/java/com/marklogic/hub/spark/sql/sources/v2/WriteDataTest.java

```diff
@@ -77,18 +96,16 @@ public void testBulkIngestWithoutUriPrefix() throws IOException {
      * @param uriPrefix
      * @return
      */
-    private DataWriter<InternalRow> buildDataWriter(String batchSize, String uriPrefix) {
+    private DataWriter<InternalRow> buildDataWriter(String batchSize, String uriPrefix, String endpointState, String workUnit, String apiPath) {
```

Comment:
When there are a lot of params in a method, that's generally a sign that a class should be created. 

While we don't need this class in the application code, an "Options" class would be very helpful here for making it easier to write tests. It would have properties of batchSize, uriPrefix, apiPath, etc. And then it would have a "Map toMap()" method that builds a Map of params based on what's been configured. 

Then, each test will just build an Options class based on the scenario that it's testing. That will make it easier to write future tests - i.e. when @SameeraPriyathamTadikonda  needs to test for collections/permissions/etc, we don't have to add those to this method and fix every method that calls it - we just add new properties to our Options class, and the existing tests won't be affected.